modules/post/multi/recon/local_exploit_suggester.rb

```diff
@@ -108,58 +164,168 @@ def show_found_exploits
-    @local_exploits.each do |m|
+    results = wanted_modules.map.with_index do |mod, index|
+      print "%bld%blu[*]%clr Running check method for exploit #{index + 1} / #{wanted_modules.count}\r"
       begin
-        checkcode = m.check
+        checkcode = mod[:module].check
       rescue => e
-        elog("#Local Exploit Suggester failed with: #{e.class} when using #{m.shortname}", error: e)
-        vprint_error "Check with module #{m.fullname} failed with error #{e.class}"
-        next
+        elog("#Local Exploit Suggester failed with: #{e.class} when using #{mod[:module].shortname}", error: e)
+        vprint_error "Check with module #{mod[:module].fullname} failed with error #{e.class}"
+        next { module: mod[:module], errors: ['The check raised an exception.'] }
       end
 
       if checkcode.nil?
-        vprint_error "Check failed with #{m.fullname} for unknown reasons"
-        next
+        vprint_error "Check failed with #{mod[:module].fullname} for unknown reasons"
+        next { module: mod[:module], errors: ['The check failed for unknown reasons.'] }
       end
 
       # See def is_check_interesting?
       unless is_check_interesting? checkcode
-        vprint_status "#{m.fullname}: #{checkcode.message}"
-        next
+        vprint_status "#{mod[:module].fullname}: #{checkcode.message}"
+        next { module: mod[:module], errors: [checkcode.message] }
       end
 
       # Prints the full name and the checkcode message for the exploit
-      print_good "#{m.fullname}: #{checkcode.message}"
-      results << [m.fullname, checkcode.message]
+      print_good "#{mod[:module].fullname}: #{checkcode.message}"
 
       # If the datastore option is true, a detailed description will show
-      next unless datastore['SHOWDESCRIPTION']
-
-      # Formatting for the description text
-      Rex::Text.wordwrap(Rex::Text.compress(m.description), 2, 70).split(/\n/).each do |line|
-        print_line line
+      if datastore['SHOWDESCRIPTION']
+        # Formatting for the description text
+        Rex::Text.wordwrap(Rex::Text.compress(m[:module].description), 2, 70).split(/\n/).each do |line|
+          print_line line
+        end
       end
+
+      next { module: mod[:module], checkcode: checkcode.message }
     end
 
+    vprint_line
+    vprint_status valid_modules_table(results)
+
+    vprint_line
+    vprint_session_info
+    vprint_status unwanted_modules_table(@local_exploits.reject { |mod| mod[:wanted] })
+
     report_note(
       :host => session.session_host,
       :type => 'local.suggested_exploits',
       :data => results
     )
   end
 
+  def valid_modules_table(results)
+    name_styler = ::Msf::Ui::Console::TablePrint::CustomColorStyler.new({})
+    check_styler = ::Msf::Ui::Console::TablePrint::CustomColorStyler.new({})
+
+    partitions = results.partition { |result| result[:checkcode] }
+    all_rows = partitions.first + partitions.second
+    rows = all_rows.map.with_index do |result, index|
```

Comment:
It's very easy to get mixed up on what's what here, i.e. `all_rows` vs `rows`, vs `results` - which is also similar to rows.

Potentially:
```suggestion
    checkcode_rows, without_checkcode_rows = results.partition { |result| result[:checkcode] }
    rows = (checkcode_rows + without_checkcode_rows).map.with_index do |result, index|
```